app/controlplane/internal/service/workflow.go

```diff
@@ -57,6 +59,11 @@ func (s *WorkflowService) Create(ctx context.Context, req *pb.WorkflowServiceCre
 		return nil, err
 	}
 
+	if err = s.userHasPermissionOnProject(ctx, s.projectsUseCase, currentOrg.ID, req.GetProjectName(), authz.PolicyWorkflowCreate); !errors.IsNotFound(err) {
```

Comment:
I do not understand this code. 

`s.userHasPermissionOnProject` will return `nil` and hence return the whole thing?

Reply:
This particular case is needed because the CreateWorkflow method is also responsible of creating projects. This way, if the project doesn't exist, we get a 404, but it's still valid in this scenario, so it must be ignored.
I'll make sure it's clear in the code.